runtime/jcl/common/system.c

```diff
@@ -70,12 +70,46 @@ jstring JNICALL Java_java_lang_System_getEncoding(JNIEnv *env, jclass clazz, jin
 	return getEncoding(env, encodingType);
 }
 
+#if JAVA_SPEC_VERSION >= 18
+jstring JNICALL
+Java_java_lang_System_getSysPropBeforePropertiesInitialized(JNIEnv *env, jclass clazz, jint sysPropID)
+{
+	const char *sysPropValue = NULL;
+	jstring result = NULL;
+	PORT_ACCESS_FROM_ENV(env);
+
+	switch (sysPropID) {
+	case 0: /* os.version */
+		/* Same logic as vmprops.c:initializeSystemProperties(vm) - j9sysinfo_get_OS_version() */
+		sysPropValue = j9sysinfo_get_OS_version();
+		if (NULL != sysPropValue) {
+#if defined(WIN32)
+			char *cursor = strchr(propValue, ' ');
```

Comment:
This should reference `sysPropValue`, not `propValue`.

Reply:
Aha, fixed.